File: Pixytech.Desktop.Infrastructure/Services/Interfaces/IDialogService.cs
namespace Pixytech.Desktop.Presentation.Infrastructure.Services.Interfaces
{
    public interface IDialogService
    {
        bool Activate(ViewModelBase viewModel);
        
        bool? ShowDialog(ViewModelBase viewModel, IDialogOptions dialogOptions);

        bool? ShowDialog(ViewModelBase viewModel);

        void Show(ViewModelBase viewModel, IDialogOptions dialogOptions);

        void Show(ViewModelBase viewModel);

        void Close(ViewModelBase viewModel, bool? dialogResult = null);
    }
}

File: sample/kinetix-core/src/Components/DialogService/IDialogService.ts

import { IViewModel } from "../../Mvvm";
import { IDialogComponent } from "./IDialogComponent";
import { IDialogOptions } from "./IDialogOptions";

export interface IDialogService {
  
  getDialog(viewModel: IViewModel):IDialogComponent | undefined;

  Activate(viewModel: IViewModel): boolean;

  ShowDialog(viewModel: IViewModel,dialogOptions?: IDialogOptions): Promise<boolean>;

  Show(viewModel: IViewModel, dialogOptions?: IDialogOptions): void;

  Close(viewModel: IViewModel,result?:boolean): void;
}
